library/common/glfw/embedder.cc

```diff
@@ -385,6 +402,26 @@ FlutterWindowRef FlutterEmbedderCreateWindow(
   return state;
 }
 
+bool FlutterEmbedderEngineShutdown(FlutterEngineRef engine_ref) {
+  std::cout << "Shutting down flutter engine process." << std::endl;
+  auto result = FlutterEngineShutdown(engine_ref->engine);
+  return (result == kSuccess);
```

Comment:
You should `delete engine_ref` before returning, so it doesn't leak the struct (even if it is tiny).